velox/connectors/hive/FileHandle.cpp

```diff
@@ -34,7 +34,7 @@ std::string groupName(const std::string& filename) {
 } // namespace
 
 std::shared_ptr<FileHandle> FileHandleGenerator::operator()(
-    const std::string& filename) {
+    const FileHandleKey& fileProperties) {
```

Comment:
Can we pass the extra properties as extra parameter (encapsulated in a `struct`) and not merging them with the path (which is the cache key)?

Reply:
The cache can still be done on just the file name, but I think we need to pass all three properties in a single struct because the` CachedFactory::generate` takes in a single key https://github.com/facebookincubator/velox/blob/main/velox/common/caching/CachedFactory.h#L137

Reply:
It does not look clean if the whole struct is used as key.  We need to carry it around separately somehow if possible.

Reply:
Do you suggest trying to maybe change the signature (if it is possible) to something like
```
template <typename Key, typename Value, typename Generator>
std::pair<bool, Value> CachedFactory<Key, Value, Generator>::generate(
    const Key& key, struct otherProperties) {
```
?

Reply:
@mbasmanova do you have any suggestions on how to split file name from other properties?

Reply:
@acvictor Yes I think that is the proper way to implement this.  The key is (path, version) and size, mtime, fd are extra properties.

Reply:
Let me try to factor, thanks!

Reply:
@Yuhta I've made changes to revert the key to the original filename string. Can you please review again?